orchagent/muxorch.cpp

```diff
@@ -821,28 +828,159 @@ MuxCable* MuxOrch::findMuxCableInSubnet(IpAddress ip)
     return nullptr;
 }
 
-bool MuxOrch::isNeighborActive(IpAddress nbr, string alias)
+bool MuxOrch::isNeighborActive(const IpAddress& nbr, const MacAddress& mac, string& alias)
 {
+    if (mux_cable_tb_.empty())
+    {
+        return true;
+    }
+
     MuxCable* ptr = findMuxCableInSubnet(nbr);
 
     if (ptr)
     {
         return ptr->isActive();
     }
 
+    string port;
+    if (!getMuxPort(mac, alias, port))
+    {
+        SWSS_LOG_INFO("Mux get port from FDB failed for '%s' mac '%s'",
+                       nbr.to_string().c_str(), mac.to_string().c_str());
+        return true;
+    }
+
+    if (!port.empty() && isMuxExists(port))
+    {
+        MuxCable* ptr = getMuxCable(port);
+        return ptr->isActive();
+    }
+
+    return true;
+}
+
+bool MuxOrch::getMuxPort(const MacAddress& mac, const string& alias, string& portName)
+{
+    portName = std::string();
+    Port rif, port;
+
+    if (!gPortsOrch->getPort(alias, rif))
```

Comment:
>rif [](start = 36, length = 3)

rif is usually router interface. is it router interface here? i though it is a normal port.

Reply:
This is in the rif context.. alias here is for e.g 'Vlan1000' or any rif where neighbor is learnt.  Actual port is obtained at line 879